index.bs

```diff
@@ -1581,9 +1618,21 @@ operates at a higher security level than the rest of the authenticator. This is
 are embedded in the WebAuthn client, as in those cases this cryptographic module (which may, for example, be a TPM) could be
 considered more trustworthy than the rest of the authenticator.
 
-Each authenticator stores some number of [=public key credentials=]. Each [=public key credential=] has an identifier which is
-unique (or extremely unlikely to be duplicated) among all [=public key credentials=]. Each credential is also associated with a
-[=[RP]=], whose identity is represented by a [=Relying Party Identifier=] ([=RP ID=]).
+Each authenticator stores a <dfn for=authenticator>credentials map</dfn>, a [=map=] from ([=Relying Party Identifier=], [=user
+handle=]) to [=public key credential source=].
+
+<div algorithm="looking up a credential id">
+
+The result of <dfn for="credential id">looking up</dfn> a [=credential id=] |credentialId| in an [=authenticator=]
+|authenticator| is the result of the following algorithm:
+1. If |authenticator| can decrypt |credentialId| into a [=public key credential source=] |credSource|:
+    1. Set |credSource|.[=public key credential source/id=] to |credentialId|.
+    1. Return |credSource|.
+1. [=map/For each=] <var ignore>key</var> → |credSource| of |authenticator|'s [=credentials map=]:
+    1. If |credSource|.[=public key credential source/id=] is |credentialId|, return |credSource|.
+1. Return `null`.
+
+</div>
```

Comment:
I suggest we keep all the gory algorithmic stuff down in subsections below this introductory prose.

Reply:
I agree.

Reply:
BTW, I continue to wish to move this algorithm to it's own section.  I did not do that as a part of the major merge-from-master catchup, but will do now if there's no objections.